improver_tests/utilities/cube_extraction/test_ExtractPressureLevel.py

```diff
@@ -102,34 +102,137 @@ def cube_shape_check_without_realizations(pressure_slice_cube):
+    temperature,
+    temperature_on_pressure_levels,
+    expected_p_index,
+    with_realization,
+    special_value,
+    reverse_pressure,
+    least_significant_digit,
 ):
     """Tests the ExtractPressureLevel plugin with values for temperature and
     temperature on pressure levels to check for expected result.
+    Tests behaviour when temperature and/or pressure increase or decrease along
+    the pressure axis.
+    Tests behaviour with different special values in the temperature data.
+    Tests behaviour with and without a realization coordinate.
     Also checks the metadata of the output cube"""
+    special_value_index = 0
+    positive_correlation = True
+    if reverse_pressure:
+        # Flip the pressure coordinate for this test. We also swap which end the
+        # special value goes, so we can test _one_way_fill in both modes.
+        temperature_on_pressure_levels.coord(
+            "pressure"
+        ).points = temperature_on_pressure_levels.coord("pressure").points[::-1]
+        special_value_index = -1
+        positive_correlation = False
+    expected = np.interp(
+        expected_p_index,
+        range(len(temperature_on_pressure_levels.coord("pressure").points)),
+        temperature_on_pressure_levels.coord("pressure").points,
+    )
     expected_data = np.full_like(
         temperature_on_pressure_levels.data[:, 0, ...], expected
     )
+
+    if special_value is True:
+        # This is a proxy for setting a mask=True entry
+        temperature_on_pressure_levels.data = np.ma.MaskedArray(
+            temperature_on_pressure_levels.data, mask=False
+        )
+        temperature_on_pressure_levels.data.mask[
+            0, special_value_index, 0, 0
+        ] = special_value
+    elif special_value is None:
+        pass
+    else:
+        temperature_on_pressure_levels.data = temperature_on_pressure_levels.data.copy()
+        if isinstance(special_value, float):
+            temperature_on_pressure_levels.data[
+                0, special_value_index, 0, 0
+            ] = special_value
+        else:
+            if special_value_index < 0:
+                temperature_on_pressure_levels.data[0, -2:, 0, 0] = special_value
+            else:
+                temperature_on_pressure_levels.data[0, 0:2, 0, 0] = special_value
+
     if not with_realization:
         temperature_on_pressure_levels = temperature_on_pressure_levels[0]
         expected_data = expected_data[0]
-    result = ExtractPressureLevel(value_of_pressure_level=temperature)(
-        temperature_on_pressure_levels
-    )
+
+    if least_significant_digit:
+        temperature_on_pressure_levels.attributes[
+            "least_significant_digit"
+        ] = least_significant_digit
+
+    result = ExtractPressureLevel(
+        value_of_pressure_level=temperature, positive_correlation=positive_correlation
+    )(temperature_on_pressure_levels)
+    assert not np.ma.is_masked(result.data)
```

Comment:
I think you've pushed parameterisation to its zenith (or nadir, depending upon perspective). Whilst I don't want you to rewrite this, there is definitely an argument for breaking tests up at some point so that the remain clear, even if it adds a few more lines. For my sanity Stephen. Think of my sanity.

Reply:
I did think about it, but actually, I wanted to be sure that each possible permutation was tested, so nesting them all into 160 unique tests seemed the best approach. It was really helpful when debugging the code!